### DISCUSSION TITLE
Accessing/exporting jinja variables

### DISCUSSION THREAD

**Discussion by jeroen-ultra** [mitsuhiko/minijinja#645] [Q&A]
Is it possible to `set` multiple variables (at the top-level scope), and (rather than using them in the template output) determine which names were set by the expression, and extract those values?

Alternatively, is it possible to alter the values of variables that were provided in the context to the expression, and retrieve the new values after the expression is evaluated?

**Comment by mitsuhiko** [comment]
I'm not sure if I understood your question correctly but I believe you are looking for `eval_to_state`.  Have a look at this example which shows how to look up a variable after evaluation: https://github.com/mitsuhiko/minijinja/blob/main/examples/eval-to-state/src/main.rs
